fuse_optimizers/include/fuse_optimizers/variable_stamp_index.h

```diff
@@ -86,13 +91,24 @@ class VariableStampIndex
     stamped_index_.clear();
     variables_.clear();
     constraints_.clear();
+    unique_stamps_.clear();
   }
 
   /**
    * @brief Returns the most recent timestamp associated with any variable
    */
   ros::Time currentStamp() const;
 
+  /**
+   * @brief Returns the oldest/first stamp 
+   */
+  ros::Time firstStamp() const;
+
+  /**
+   * @brief Returns the ith timestamp
+   */
+  ros::Time ithStamp(size_t idx) const;
```

Comment:
If you need access to the first, last, and i-th element of the unique stamps set, would it make more sense just to provide read-only access to the container itself?
```
const std::set<ros::Time>& stamps() const { return unique_stamps_; }
```

Reply:
I was thinking of overriding the [] operator to perform the function of these 3 functions instead, this way it keeps the validity checking to the stamp index rather than to the client using it